### PR TITLE
tests: create test for dynamic loadable logging backends

### PR DESCRIPTION
**Pull request by shuai1x**
Test the three APIs, log_backend_activate,
log_backend_is_active and log_backend_deactivate.

Signed-off-by: Zhao Shuai <shuai1x.zhao@intel.com>

### CODE REVIEW

**Review by nashif — APPROVED** [review]

**Review by teburd — APPROVED** [review]

**Comment by laurenmurphyx64** [comment]
Apologies to @nordic-krch, I'm new as a release manager so didn't realize we needed assignee approval to merge :sweat_smile: Let me know if you have issues with this PR and we can work to submit changes / undo it.

**Pull request merged**
